runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java

```diff
@@ -70,6 +106,64 @@ public void updateClanChatTitle()
 		}
 	}
 
+	private boolean send(ChatboxInput chatbox)
+	{
+		executor.execute(() ->
+		{
+			chatbox.resume();
+		});
+
+		return true;
+	}
+
+	private boolean isClanChatClosed()
+	{
+		Widget widget = client.getWidget(WidgetInfo.CLAN_CHAT);
+
+		if (widget == null)
+		{
+			return true;
+		}
+
+		return widget.isHidden();
+	}
+
+	@Override
+	public boolean onChatboxInput(ChatboxInput chatboxInput)
+	{
+		if (!config.swapChat() && (!config.mobileChat() || isClanChatClosed()))
+		{
+			return send(chatboxInput);
+		}
+
+		String msg = chatboxInput.getValue();
+
+		if (msg.startsWith("/"))
+		{
+			msg = msg.substring(1);
+			chatboxInput.setChatType(0);
+		}
+		else
+		{
+			msg = "/" + msg;
+			chatboxInput.setChatType(2);
+		}
+
+		chatboxInput.setValue(msg);
+
+		return send(chatboxInput);
+	}
+
+	@Override
+	public boolean onPrivateMessageInput(PrivateMessageInput privateMessageInput)
+	{
+		executor.execute(() ->
```

Comment:
Same here, you just do not need to do anything in this method

Reply:
It's abstract, have to have it.

Reply:
Yeah, but I mean the method does not need body